src/content/app/entity-viewer/state/api/queries/proteinDomainsQuery.ts

```diff
@@ -0,0 +1,58 @@
+/**
+ * See the NOTICE file distributed with this work for additional information
+ * regarding copyright ownership.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+import { gql } from 'graphql-request';
+import { Pick2, Pick3 } from 'ts-multipick';
+
+import type { FamilyMatches } from 'src/shared/types/thoas/product';
+
+// A light query to get brief information about a gene.
+// Useful for populating the top bar and example links in the interstitial.
+export const proteinDomainsQuery = gql`
+  query ProteinDomains($genomeId: String!, $productId: String!) {
+    product(genome_id: $genomeId, stable_id: $productId) {
+      length
+      family_matches {
+        relative_location {
+          start
+          end
+        }
+        sequence_family {
+          name
+          description
+          source {
+            name
+          }
+        }
+      }
+    }
+  }
+`;
+
+export type FamilyMatchesProduct = Pick2<
+  FamilyMatches,
+  'relative_location',
+  'start' | 'end'
+> &
+  Pick2<FamilyMatches, 'sequence_family', 'name' | 'description'> &
+  Pick3<FamilyMatches, 'sequence_family', 'source', 'name'>;
+
+export type ProteinDomainsQueryResult = {
+  product: {
+    length: number;
+    family_matches: FamilyMatchesProduct[];
+  };
```

Comment:
Also need to change to `FamilyMatch` in the above four places as well in the code

Reply:
Done

Reply:
I still find this difficult to read. To me, the name of the `FamilyMatchProduct` type should be `FamilyMatchInProduct` (notice the word "in"). At the very least, if you don't like the "in", it should be `ProductFamilyMatch`, although this looks more cumbersome. `FamilyMatchProduct` reads as if it is a product of a family match, which isn't right.

Reply:
And i understood it as a product of family match. if it is actually a product in family match then makes sense to change it.

Reply:
> i understood it as a product of family match. if it is actually a product in family match

It's neither.

A "product" in this context is a protein. And family matches are the matches that this protein has within a protein family (e.g. pfam), or something like that. Each match corresponds to what we call a protein domain.